L1Trigger/L1CaloTrigger/plugins/L1EGammaEEProducer.cc

```diff
@@ -0,0 +1,190 @@
+
+      // cl3d->pt()/(1+cl3d->hOverE())
+
+      // this is not yet used
+      used_clusters.insert(cl3d);
+      auto eta_phi_bin = get_eta_phi_bin(cl3d);
+      // std::cout << " cl pt: " << cl3d->pt()
+      //           << " eta: " << cl3d->eta()
+      //           << " phi: " << cl3d->phi()
+      //           << " eta, phi bin: " << eta_phi_bin.first << "," << eta_phi_bin.second
+      //           << std::endl;
+
+      for (int eta_bin : {eta_phi_bin.first - 1, eta_phi_bin.first, eta_phi_bin.first + 1}) {
+        for (int phi_bin : {eta_phi_bin.second - 1, eta_phi_bin.second, eta_phi_bin.second + 1}) {
+          auto bucket = etaphi_bins.find(std::make_pair(eta_bin, phi_bin));
+          if (bucket != etaphi_bins.end()) {
+            // this bucket is not empty
+            for (auto other_cl_ptr : bucket->second) {
+              if (used_clusters.find(other_cl_ptr) == used_clusters.end()) {
+                if (fabs(other_cl_ptr->eta() - cl3d->eta()) < 0.02) {
+                  if (fabs(reco::deltaPhi(other_cl_ptr->phi(), cl3d->phi())) < 0.1) {
+                    // std::cout << "MERGE with cl pt: " << other_cl_ptr->pt()
+                    //           << " eta: " << other_cl_ptr->eta()
+                    //           << " phi: " << other_cl_ptr->phi()
+                    //           <<  std::endl;
+                    float pt_other = other_cl_ptr->pt();
+                    if (other_cl_ptr->hOverE() != -1)
+                      pt_other = other_cl_ptr->pt() / (1 + other_cl_ptr->hOverE());
+                    mom += reco::Candidate::PolarLorentzVector(pt_other, other_cl_ptr->eta(), other_cl_ptr->phi(), 0.);
+                    mom_eint += reco::Candidate::PolarLorentzVector(
+                        other_cl_ptr->iPt(l1t::HGCalMulticluster::EnergyInterpretation::EM),
+                        other_cl_ptr->eta(),
+                        other_cl_ptr->phi(),
+                        0.);
+                    used_clusters.insert(other_cl_ptr);
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+      float calib_factor = calibrator_.calibrationFactor(mom.pt(), mom.eta());
+      l1t::EGamma eg =
+          l1t::EGamma(reco::Candidate::PolarLorentzVector(mom.pt() / calib_factor, mom.eta(), mom.phi(), 0.));
+      // l1t::EGamma eg(mom);
+      // FIXME: full duplication with HWqual 2
+      eg.setHwQual(3);
+      eg.setHwIso(1);
+      l1EgammaBxCollection->push_back(0, eg);
+
+      l1t::EGamma eg_emint_brec =
+          l1t::EGamma(reco::Candidate::PolarLorentzVector(mom_eint.pt(), mom_eint.eta(), mom_eint.phi(), 0.));
+      // l1t::EGamma eg(mom);
+      // FIXME: full duplication with HWqual 2
+      eg_emint_brec.setHwQual(5);
+      eg_emint_brec.setHwIso(1);
+      l1EgammaBxCollection->push_back(0, eg_emint_brec);
+    }
+  }
+
+  iEvent.put(std::move(l1EgammaBxCollection), "L1EGammaCollectionBXVWithCuts");
```

Comment:
is this tag needed since the producer doesn't produce anything else?

Reply:
removed. done.